mllib/src/main/scala/org/apache/spark/ml/util/ReadWrite.scala

```diff
@@ -282,7 +282,10 @@ private[ml] object DefaultParamsReader {
    */
   def loadMetadata(path: String, sc: SparkContext, expectedClassName: String = ""): Metadata = {
     val metadataPath = new Path(path, "metadata").toString
-    val metadataStr = sc.textFile(metadataPath, 1).first()
+    val metadataStr = try sc.textFile(metadataPath, 1).first() catch {
+      case _: UnsupportedOperationException =>
+        throw new IllegalArgumentException(s"Empty metadata for path $path")
+    }
```

Comment:
Or
```scala
    val textFile = sc.textFile(metadataPath, 1)
    if (textFile.isEmpty()) throw new IllegalArgumentException(s"Empty metadata for path $path")
    val metadataStr = textFile.first()
```
But I thought calling twice `take(1)` threw `isEmpty()` and `first()` would be a waist.